File: Maze.py
N = 12
M = 12
i = 2
j = 0
grid = [['#', '#', '#', '#', '#', '#', '#', '#', '#', '#', '#', '#'],
        ['#', '.', '.', '.', '#', '.', '.', '.', '.', '.','.','#'],
        ['.', '.', '#', '.', '#', '.', '#', '#', '#', '#', '.', '#'],
        ['#', '#', '#', '.', '#','.', '.', '.', '.', '#', '.', '#'],
        ['#', '.', '.', '.', '.', '#', '#', '#', '.', '#', '.', '#'],
        ['#', '#', '#', '#', '.', '#', '.', '#', '.', '#', '.', '.'],
        ['#', '.', '.', '#', '.', '#', '.', '#', '.', '#', '.', '#'],
        ['#', '#', '.', '#', '.', '#', '.', '#', '.', '#', '.', '#'],
        ['#', '.', '.', '.', '.', '.', '.', '.', '.', '#', '.', '#'],
        ['#', '#', '#', '#', '#', '#', '.', '#', '#', '#', '.', '#'],
        ['#', '.', '.', '.', '.', '.', '.', '#', '.', '.', '.', '#'],
        ['#', '#', '#', '#', '#', '#', '#', '#', '#', '#', '#', '#']]
def clear_game():
    grid[2][0] = 'X'
    grid[5][11] = '.'
def check_input(c):
    if(c == 8 or c == 4 or c == 5 or c == 6):
        return True
    return False
def valid_move(c):
    global grid
    global i
    global j
    if(c == 8):
        if(grid[i-1][j] == '.'):
            return True
    elif(c == 4):
        if(grid[i][j-1] == '.'):
            return True
    elif(c== 6):
        if(grid[i][j+1] == '.'):
            return True
    elif(c == 5):
        if(grid[i+1][j] == '.'):
            return True
    return False
def set_position(c):
     global grid
     global i
     global j
     grid[i][j] = '.'
     if(c == 8):
        grid[i-1][j] = 'X'
        i = i-1
     elif(c == 4):
        grid[i][j-1] = 'X'
        j = j - 1
     elif(c == 6):
        grid[i][j+1] = 'X'
        j = j + 1
     elif(c == 5):
        grid[i+1][j] = 'X'
        i = i + 1
def print_grid():
    global grid
    for i in range(N):
        for j in range(M):
            print (grid[i][j], end = ' ')
        print()
def read_input():
    print("----------------")
    print("Enter the position: ", end = ' ')
    c = int(input())
    while(not valid_move(c) or not check_input(c)):
        print("Enter valid number: ",end = ' ')
        c = int(input())
    return c
def check_win():
    global i
    global j
    if(i == 5 and j == 11):
        return True
def play():
    c = 0
    while True:
        print_grid()
        c = read_input()
        set_position(c)
        if(check_win()):
            print_grid()
            print("Winner!!")
            break
while True:
    print("Welcome to the Maze Game.....")
    while True:
      clear_game()
      play()
      c = input('Play Again [Y/N] ')
      if c not in 'yY':
        break

File: Maze.cpp
#include <bits/stdc++.h>
using namespace std;
int i = 2;
int j = 0;
const int N = 12;
char grid[N][N] =
      { { '#', '#', '#', '#', '#', '#', '#', '#', '#', '#', '#', '#' },
		{ '#', '.', '.', '.', '#', '.', '.', '.', '.', '.','.','#' },
		{ '.', '.', '#', '.', '#', '.', '#', '#', '#', '#', '.', '#' },
		{ '#', '#', '#', '.', '#','.', '.', '.', '.', '#', '.', '#' },
		{ '#', '.', '.', '.', '.', '#', '#', '#', '.', '#', '.', '#' },
		{ '#', '#', '#', '#', '.', '#', '.', '#', '.', '#', '.', '.' },
		{ '#', '.', '.', '#', '.', '#', '.', '#', '.', '#', '.', '#' },
		{ '#', '#', '.', '#', '.', '#', '.', '#', '.', '#', '.', '#' },
		{ '#', '.', '.', '.', '.', '.', '.', '.', '.', '#', '.', '#' },
		{ '#', '#', '#', '#', '#', '#', '.', '#', '#', '#', '.', '#' },
		{ '#', '.', '.', '.', '.', '.', '.', '#', '.', '.', '.', '#' },
		{ '#', '#', '#', '#', '#', '#', '#', '#', '#', '#', '#', '#' } };
void clear_game()
{
    grid[2][0] = 'X';
    grid[5][11] = '.';
}
void print_grid()
{
    for(int i = 0; i < N; i++)
    {
        for(int j = 0; j < N; j++)
        {
            cout << grid[i][j] << " ";
        }
        cout << endl;
    }
}
bool check_input(int input)
{
    if(input == 8 || input == 4 || input == 5 || input == 6)
        return true;
    return false;
}
bool valid_move(int input)
{
    if(input == 8)
    {
        if(grid[i-1][j] == '.')
            return true;
    }
    else if(input == 4)
    {
        if(grid[i][j-1] == '.')
            return true;
    }
    else if(input == 6)
    {
        if(grid[i][j+1] == '.')
            return true;
    }
    else if(input == 5)
    {
        if(grid[i+1][j] == '.')
            return true;
    }
    return false;
}
void read_input(int& input)
{
    cout << "----------------" << endl;
    cout << "Enter the position: ";
    cin >> input;
    while(!valid_move(input) || !check_input(input))
    {
        cout << "Enter valid number: ";
        cin >> input;
    }
}
void set_position(int input)
{
     grid[i][j] = '.';
     if(input == 8)
     {
        grid[i-1][j] = 'X';
        i = i-1;
     }
     else if(input == 4)
     {
        grid[i][j-1] = 'X';
        j = j - 1;
     }
    else if(input == 6)
    {
        grid[i][j+1] = 'X';
        j = j + 1;
    }
    else if(input == 5)
    {
        grid[i+1][j] = 'X';
        i = i + 1;
    }
}
bool check_win()
{
    if(i == 5 && j == 11)
        return true;
    return false;
}
void play()
{
    while(true)
    {
        print_grid();
        int input;
        read_input(input);
        set_position(input);
        if(check_win())
        {
            print_grid();
            cout << "Winner!!" << endl;
            break;
        }
    }
}
int main()
{
    cout<<"Welcome to the Maze Game....."<<endl;
    while(true)
    {
      clear_game();
      play();
      char c;
      cout<<"Play Again [Y/N]";
      cin>>c;
      if(c != 'y' && c!= 'Y')
        break;
    }
}
